### PR TITLE
We now incrementally calculate the queue delivery time

### PR DESCRIPTION
**Pull request by annando**
Currently we really penetrated other systems with the redelivering of messages (every 15 minutes for 12 hours, then for two days every hour). This was bad for the performance of the delivering system, but could also had been bad for the receiver if it had got load problems.

We are now using the mechanism that Diaspora is using as well. We try 14 times with an increasing time between each trial.

### CODE REVIEW

File: src/Model/Queue.php [hunk 1]
```diff
@@ -19,7 +19,22 @@ class Queue
...
+
+		$retrial = $queue['retrial'];
+
+		if ($retrial > 14) {
+			self::removeItem($id);
+		}
+
+		// Calculate the delay until the next trial
+		$delay = (($retrial + 3) ** 4) + (rand(1, 30) * ($retrial + 1));
+		$next = DateTimeFormat::utc(date('c', time() + $delay));
```
Comment (MrPetovan) [inline src/Model/Queue.php]: You didn't have to call `date()`, you can use the `"now + $delay seconds"` string with `DateTimeFormat::utc()` directly.

Comment (annando) [reply src/Model/Queue.php]: I didn't knew.

Comment (annando) [reply src/Model/Queue.php]: Done.

Comment (MrPetovan) [reply src/Model/Queue.php]: "I didn't know", no need to use the past tense you already used with "didn't". I 'm still making the mistake often.

File: src/Model/Queue.php [hunk 2]
```diff
@@ -32,7 +32,7 @@ public static function updateTime($id)
 
 		// Calculate the delay until the next trial
 		$delay = (($retrial + 3) ** 4) + (rand(1, 30) * ($retrial + 1));
-		$next = DateTimeFormat::utc(date('c', time() + $delay));
+		$next = DateTimeFormat::utc('now ' . $delay . ' seconds');
```
Comment (MrPetovan) [inline src/Model/Queue.php]: You forgot the `+` in the string format.

Comment (annando) [reply src/Model/Queue.php]: Oops ...

**Comment by annando** [comment]
Concerning the delay times: Please consider that the queue is started from the cron - that is started only every few minutes.

**Comment by MrPetovan** [comment]
That's what I mean, the first retrial delay isn't really a delay since it's going to be picked up on the next worker execution. But do we really want that for performance purposes? If we push the delay to 2 cron executions instead, it will ease off the load pressure on our cronjobs.

**Comment by annando** [comment]
Per standard the cron interval is set to 10 minutes. So 0 to 2 will be after 10 minutes. So the times in fact are these:
- 0: 10 minutes
- 1: 10 minutes
- 2: 10 minutes
- 3: 20 minutes
- 4: 40 minutes
- 5: 60-70 minutes
- 6: 100-110 minutes
...

**Comment by MrPetovan** [comment]
Indeed, I have it set for 5 minutes interval but the defaults is indeed 10. Still happy with the scale?

**Comment by annando** [comment]
I am, since the queue mechanism is started with the priority ```PRIORITY_NEGLIGIBLE``` - which in fact means that it is only executed when the system isn't busy with other stuff.

**Pull request merged**